python/pyarrow/parquet.py

```diff
@@ -1479,6 +1479,11 @@ def read(self, columns=None, use_threads=True, use_pandas_metadata=False):
                 ]
                 columns = columns + list(set(index_columns) - set(columns))
 
+        if len(list(self._dataset.get_fragments())) <= 1:
+            # Allow per-column parallelism; would otherwise cause contention
+            # in the presence of per-file parallelism.
+            use_threads = False
```

Comment:
We should probably check that `use_threads`is actually True when enabling this per-column parallelism

Reply:
Alternative would also to handle this within the `to_table` method?